extensions/metadata_exchange/plugin.h

```diff
@@ -71,7 +72,9 @@ class PluginRootContext : public RootContext {
 // Per-stream context.
 class PluginContext : public Context {
  public:
-  explicit PluginContext(uint32_t id, RootContext* root) : Context(id, root) {}
+  explicit PluginContext(uint32_t id, RootContext* root) : Context(id, root) {
+    direction_ = ::Wasm::Common::getTrafficDirection();
```

Comment:
Is this just as efficient as getting during config time?
This call will be made for every request? Why not make the filter direction aware just like everything else?

Reply:
It feels expensive to run two VMs just to pick one based on the listener direction. I'm not sure how much more expensive adding this extra ABI call is.